Wallet/Cryptography/SqlConnector.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Data.Common;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+using Wallet.Communication;
+
+namespace Wallet.Cryptography
+{
+    public class SqlConnector : IDatabase
+    {
+        private SqlConnectionStringBuilder m_sqlConnectionStringBuilder;
+        private bool m_isInitialized;
+
+        // SQL queries
+        private const string CreateAccountsTableQuery = @"
+                            If not exists (select name from sysobjects where name = 'accounts')
+                            CREATE TABLE accounts
+                            (
+                               Id  nchar(30) not null 
+                                  PRIMARY KEY,
+                               PrivateKey  nvarchar(128)     not null
+                            );";
+
+        private const string GetPrivateKeyByIdQueryTempalte = @"SELECT * FROM accounts Where Id='{0}'";
```

Comment:
a better implementation would be to save these queries as stored procedures and call them with the relevant parameters. generally performance is better since the stored procedure are "compiled" once when deploying them in the database before the execution. When sending a query from the code like you did here- the execution plane phase of the query runs each time the database is queried

Reply:
The question is, if you can created and 'deploy' a stored procedure via code /arm template

Reply:
I'm sure that you'll be able to find easily a way to do that. I'm sure that you can also run a CREATE script to create the stored procedure like you create the table if it doesn't exists.